offload/plugins-nextgen/common/CMakeLists.txt

```diff
@@ -7,7 +7,10 @@ add_library(PluginCommon OBJECT
   src/RPC.cpp
   src/Utils/ELF.cpp
 )
-add_dependencies(PluginCommon intrinsics_gen LLVMProfileData)
+add_dependencies(PluginCommon intrinsics_gen)
+if (TARGET LLVMProfileData)
+  add_dependencies(PluginCommon LLVMProfileData)
+endif()
```

Comment:
Can we remove this and move it with the others?

Reply:
I'm sorry but I don't understand the "move it with the others" part. I don't really understand why an explicit dependency was added in the first place — I thought linking against the library implies that.

Reply:
I don't think it needs to be here, it's a library target so it should go with the other LLVM libraries. Moving it should avoid the need for handling object libraries like this.

Reply:
You mean into `supported_jit_targets`?

Reply:
`add_target_library` or whatever it's called. The JIT library linking should probably also be moved there. Looking at it now, I'm not sure how it works since it's doing a private `link_libraries` on an object library, which has no link step.